src/core_modules/capture-core/components/WorkingLists/TeiWorkingLists/TrackedEntityBulkActions/Actions/CompleteAction/CompleteAction.js

```diff
@@ -0,0 +1,231 @@
+                        borderless
+                        header={i18n.t('Details (Advanced)')}
+                    >
+                        <span className={classes.errorContainer}>
+                            <ul>
+                                {errors ? errors.map(errorReport => (
+                                    <li key={`${errorReport.uid}-${errorReport.errorCode}`}>
+                                        {errorReport?.message}
+                                    </li>
+                                )) : (
+                                    <li>
+                                        {i18n.t('An unknown error occurred.')}
+                                    </li>
+                                )}
+                            </ul>
+                        </span>
+                    </Widget>
+                </div>
+            );
+        }
+
+        if (errorFetchingTrackedEntities) {
+            return (
+                <div className={classes.container}>
+                    {i18n.t('An unexpected error occurred while fetching the enrollments. Please try again.')}
+                </div>
+            );
+        }
+
+        // If there are no active enrollments, show a message and disable the complete button
+        if (enrollmentCounts.active === 0) {
+            return (
+                <div className={classes.container}>
+                    {i18n.t('There are currently no active enrollments in the selection. All enrollments are already completed or cancelled.')}
+                </div>
+            );
+        }
+
+        return (
+            <div className={classes.container}>
+                {i18n.t('This action will complete {{count}} active enrollment in your selection.',
+                    {
+                        count: enrollmentCounts.active,
+                        defaultValue: 'This action will complete {{count}} active enrollment in your selection.',
+                        defaultValue_plural: 'This action will complete {{count}} active enrollments in your selection.',
+                    })
+                }
+
+                {' '}
+
+                {enrollmentCounts.completed > 0 &&
+                    i18n.t('{{count}} enrollment already marked as completed will not be changed.', {
+                        count: enrollmentCounts.completed,
+                        defaultValue: '{{count}} enrollment already marked as completed will not be changed.',
+                        defaultValue_plural: '{{count}} enrollments already marked as completed will not be changed.',
+                    })
+                }
+
+                <Checkbox
+                    label={i18n.t('Mark all events within enrollments as complete')}
+                    checked={completeEvents}
+                    onChange={() => setCompleteEvents(prevState => !prevState)}
+                />
+
+            </div>
+        );
+    };
+
+    return (
+        <>
+            <Button
```

Comment:
For both completing and deleting enrollments it is necessary to verify the program data write access. Can you add the checks with ConditionalTooltip? 🙏